### PR TITLE
Add an isButton flag to the Semantic widget,use it for MaterialButton

### PR DESCRIPTION
**Pull request by amirh**
Will follow-up with another CL for Cupertino buttons.

This can only land after we roll the engine with https://github.com/flutter/engine/pull/4254
(Expect tests to fail until the roll)

### CODE REVIEW

File: packages/flutter/lib/src/rendering/proxy_box.dart [hunk 1]
```diff
@@ -3224,6 +3226,17 @@ class RenderSemanticsAnnotations extends RenderProxyBox {
     markNeedsSemanticsUpdate(onlyLocalUpdates: (value != null) == hadValue);
   }
 
+  /// If non-null, sets the [SemanticsNode.isButton] semantic the given value.
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: I think this documentation may have a grammar error in it.

File: packages/flutter/lib/src/widgets/basic.dart [hunk 2]
```diff
@@ -4477,6 +4478,9 @@ class Semantics extends SingleChildRenderObjectWidget {
   /// all other tabs are unselected.
   final bool selected;
 
+  /// If non-null, indicates that this subtree represents a button.
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/basic.dart]: this should probably elaborate on what "a button" means.

**Comment by Hixie** [comment]
LGTM modulo some minor documentation issues

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/rendering/proxy_box.dart (hunk 1 above)
Comment (amirh) [reply packages/flutter/lib/src/rendering/proxy_box.dart]: done

File: packages/flutter/lib/src/widgets/basic.dart (hunk 2 above)
Comment (amirh) [reply packages/flutter/lib/src/widgets/basic.dart]: I'm not sure how to elaborate what a button means, I could try to describe it as something tap-able,  but not everything that's tap-able should have this flag set to true.

Instead I added some documentation of how this affects TalkBack/VoiceOver (provides a "button" hint)

**Comment by amirh** [comment]
Will merge once an engine roll lands

**Comment by Hixie** [comment]
We should also add these semantics to the cupertino buttons.

**Pull request merged**